### PR TITLE
SAK-40437 GradebookNG GradingService refactored and centralized

### PR DESCRIPTION
**Pull request by adrianfish**
https://sakaiproject.atlassian.net/browse/SAK-40437

This commit merges the 3 grading interfaces into one, GradingService. It
also adds a load of tests and adds a lot of typing to the various
collections used.

### CODE REVIEW

File: assignment/tool/src/java/org/sakaiproject/assignment/tool/AssignmentToolUtils.java [hunk 1]
```diff
@@ -419,15 +415,13 @@ public void gradeSubmission(AssignmentSubmission submission, String gradeOption,
...
             String associateGradebookAssignment = a.getProperties().get(PROP_ASSIGNMENT_ASSOCIATE_GRADEBOOK_ASSIGNMENT);
 
-            if (!"remove".equals(gradeOption)) {
-                // update grade in gradebook
-                alerts.addAll(integrateGradebook(options, aReference, associateGradebookAssignment, null, null, null, -1, null, sReference, "update", -1));
-            } else {
-                //remove grade from gradebook
-                alerts.addAll(integrateGradebook(options, aReference, associateGradebookAssignment, null, null, null, -1, null, sReference, "remove", -1));
-            }
+            System.out.println(aReference);
```
Comment (ern) [inline assignment/tool/src/java/org/sakaiproject/assignment/tool/AssignmentToolUtils.java:418]: These should be removed or changed to log messages

File: basiclti/basiclti-portlet/pom.xml [hunk 2]
```diff
@@ -72,11 +76,6 @@
         <dependency>
             <groupId>org.springframework</groupId>
             <artifactId>spring-web</artifactId>
-        </dependency>
-        <dependency>
-            <groupId>org.sakaiproject.edu-services.gradebook</groupId>
-            <artifactId>gradebook-service-api</artifactId>
-            <scope>provided</scope>
```
Comment (ern) [inline basiclti/basiclti-portlet/pom.xml:79]: remove this scope

File: gradebookng/api/pom.xml [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+		<version>23-SNAPSHOT</version>
+    </parent>
+
+    <packaging>jar</packaging>
+
+    <properties>
+        <deploy.target>shared</deploy.target>
+    </properties>
+
+	<dependencies>
```
Comment (ern) [inline gradebookng/api/pom.xml:20]: formatting

File: gradebookng/api/src/main/java/org/sakaiproject/grading/api/GradingAuthz.java [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+
+import java.util.List;
+import java.util.Map;
+
+import org.sakaiproject.section.api.coursemanagement.CourseSection;
+
+/**
+ * Facade to external role and authorization service.
+ */
+public interface GradingAuthz {
```
Comment (ern) [inline gradebookng/api/src/main/java/org/sakaiproject/grading/api/GradingAuthz.java:27]: Need types on the api

File: gradebookng/impl/pom.xml [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+
+	<parent>
+		<groupId>org.sakaiproject.gradebookng</groupId>
+		<artifactId>gradebookng</artifactId>
+		<version>23-SNAPSHOT</version>
+    </parent>
+
+    <packaging>sakai-component</packaging>
+
+	<dependencies>
```
Comment (ern) [inline gradebookng/impl/pom.xml:16]: spacing

File: gradebookng/impl/src/main/webapp/WEB-INF/components.xml [hunk 6]
```diff
@@ -0,0 +1,115 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<beans xmlns="http://www.springframework.org/schema/beans"
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+    xsi:schemaLocation="http://www.springframework.org/schema/beans http://www.springframework.org/schema/beans/spring-beans.xsd">
+
+    <bean id="org.sakaiproject.util.ResourceLoader.grading"
+          class="org.sakaiproject.util.ResourceLoader">
+    </bean>
+
+	<bean id="gradingHibernateMappings"
```
Comment (ern) [inline gradebookng/impl/src/main/webapp/WEB-INF/components.xml:10]: bean id should be org.sakaiproject.springframework.orm.hibernate.impl.AdditionalHibernateMappingsImpl.gradebook

**Review by ern — APPROVED** [review]

File: gradebookng/impl/src/main/java/org/sakaiproject/grading/impl/] [hunk 7]
```diff
@@ -0,0 +1,51 @@
+/**
```
Comment (jesusmmp) [inline gradebookng/impl/src/main/java/org/sakaiproject/grading/impl/]:1]: Remove this file

Comment (adrianfish) [reply gradebookng/impl/src/main/java/org/sakaiproject/grading/impl/]:1]: Bit late to the party, my man :) Good spot though. I'll remove that the next time I put a PR in around this area. Thanks!

Comment (adrianfish) [reply gradebookng/impl/src/main/java/org/sakaiproject/grading/impl/]:1]: I agree. I could just do a NOJIRA PR i suppose

**Pull request merged**